internal/introspection/router.go

```diff
@@ -24,26 +24,102 @@ import (
 	"go.uber.org/yarpc/api/transport"
 )
 
-// Procedure represent a registered procedure on a dispatcher.
-type Procedure struct {
-	Name      string `json:"name"`
-	Encoding  string `json:"encoding"`
-	Signature string `json:"signature"`
-	RPCType   string `json:"rpcType"`
-}
-
 // IntrospectProcedures is a convenience function that translate a slice of
 // transport.Procedure to a slice of introspection.Procedure. This output is
 // used in debug and yarpcmeta.
-func IntrospectProcedures(routerProcs []transport.Procedure) []Procedure {
+func IntrospectProcedures(routerProcs []transport.Procedure) Procedures {
 	procedures := make([]Procedure, 0, len(routerProcs))
 	for _, p := range routerProcs {
+		var spec interface{}
+		switch p.HandlerSpec.Type() {
+		case transport.Unary:
+			spec = p.HandlerSpec.Unary()
+		case transport.Oneway:
+			spec = p.HandlerSpec.Oneway()
+		}
+		var IDLEntryPoint *IDLModule
+		if spec != nil {
+			if i, ok := spec.(IntrospectableHandler); ok {
+				if i := i.Introspect(); i != nil {
+					IDLEntryPoint = i.IDLEntryPoint
+				}
+			}
+		}
 		procedures = append(procedures, Procedure{
-			Name:      p.Name,
-			Encoding:  string(p.Encoding),
-			Signature: p.Signature,
-			RPCType:   p.HandlerSpec.Type().String(),
+			Name:          p.Name,
+			Encoding:      string(p.Encoding),
+			Signature:     p.Signature,
+			RPCType:       p.HandlerSpec.Type().String(),
+			IDLEntryPoint: IDLEntryPoint,
 		})
 	}
 	return procedures
 }
+
+// Procedure represent a registered procedure on a dispatcher.
+type Procedure struct {
+	Name          string     `json:"name"`
+	Encoding      string     `json:"encoding"`
+	Signature     string     `json:"signature"`
+	RPCType       string     `json:"rpcType"`
+	IDLEntryPoint *IDLModule `json:"idlEntryPoint,omitempty"`
+}
+
+// Procedures is a slice of Procedure.
+type Procedures []Procedure
```

Comment:
I really don't like doing this in general, I know you have it in other PRs too - over-typing in golang leads to a lot of pain in the butt situations. The primary thing is readability here - when I see `Procedures`, I don't know if it's a struct, a typedef (like it is here), etc. Doing `[]Procecure` explicitly makes the code easier to understand.